src/Test/L0/Util/WhichUtilL0.cs

```diff
@@ -89,5 +89,42 @@ public void WhichHandleFullyQualifiedPath()
                 Assert.Equal(gitPath, gitPath2);
             }
         }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Common")]
+        public void WhichThrowsWhenSymlinkBroken()
+        {
+            using TestHostContext hc = new TestHostContext(this);
+            // Arrange
+            Tracing trace = hc.GetTrace();
+            string oldValue = Environment.GetEnvironmentVariable(PathUtil.PathVariable);
+            string newValue = oldValue + Path.GetTempPath();
+            Environment.SetEnvironmentVariable(PathUtil.PathVariable, newValue);
+
+            string brokenSymlink = Path.GetTempPath() + "broken-symlink.exe";
+            string target = "no-such-file-cf7e351f";
+            File.CreateSymbolicLink(brokenSymlink, target);
+
+            // Act.
+            try
```

Comment:
Generally it's best to avoid try/catch structures in tests.

You can instead assert on the exception type first, extract the exception itself, then further assert on its message / filename.

`ArgumentException exception = Assert.Throws<FileNotFoundException>(WhichUtil.Which(...));`
`Assert.Equal("...", exception.Message);`